src/CodeGen_LLVM.cpp

```diff
@@ -1471,7 +1471,12 @@ void CodeGen_LLVM::visit(const Mod *op) {
         int bits;
         if (is_const_power_of_two_integer(op->b, &bits)) {
             Expr one = make_one(op->b.type());
-            value = builder->CreateAnd(codegen(op->a), codegen(op->b - one));
+            const Variable *v = op->a.as<Variable>();
+            if (v && ends_with(v->name, ".host"))
```

Comment:
Rather than do this with the Mod IR node, I think it would be preferable to generate a cast to uint64 in the assert stmt, and maybe handle that kind of cast instead.